src/robot/running/context.py

```diff
@@ -39,12 +41,19 @@ def namespaces(self):
         return (context.namespace for context in self)
 
     def start_suite(self, suite, namespace, output, dry_run=False):
-        ctx = _ExecutionContext(suite, namespace, output, dry_run)
+        if self._contexts:
+            self.event_loop_ref = self._contexts[-1]._event_loop
+        ctx = _ExecutionContext(suite, namespace, output, dry_run, self.event_loop_ref)
         self._contexts.append(ctx)
         return ctx
 
     def end_suite(self):
-        self._contexts.pop()
+        ctx = self._contexts.pop()
+        if not self.event_loop_ref and ctx._event_loop:
+            # handle event loop being created after ctx creation
+            self.event_loop_ref = ctx._event_loop
+        if not self._contexts and self.event_loop_ref:
+            self.event_loop_ref.close()
```

Comment:
Wouldn't it be enough to have this instead:
```python
ctx = self._contexts.pop()
if not self._contexts and ctx._event_loop:
    ctx.event_loop.close()
```

Reply:
This unfortunately doesn't close the loop correctly, i tried a different aproach now. Instead of passing the reference directly, i made a class to handle things. Tell me what you think when you check the commit.